lib/tests/test_logsource.c

```diff
@@ -415,4 +417,92 @@ Test(log_source, test_mangle_callback)
   test_source_destroy(source);
 }
 
+
+static DynamicWindowPool *
+test_dynamic_window_pool_init(gsize pool_size)
+{
+  DynamicWindowPool *pool = dynamic_window_pool_new(pool_size);
+  pool->balanced_window = pool_size;
+  dynamic_window_pool_init(pool);
+
+  return pool;
```

Comment:
Can we assert for `pool->pool_size == pool_size`?

Reply:
I'll add a new test case for this. I don't want to assert in this function.

Reply:
Update: Actually, the assert belongs to a separate unit; I want to test `LogSource` here, not the pool. The existence of this non-mocked object here is unfortunate, I agree with that.